services/factorio-mod-portal/factorio-mod-portal.service.js

```diff
@@ -0,0 +1,198 @@
+import Joi from 'joi'
+import { BaseJsonService } from '../index.js'
+import { age } from '../color-formatters.js'
+import { formatDate } from '../text-formatters.js'
+import { renderDownloadsBadge } from '../downloads.js'
+import { renderVersionBadge } from '../version.js'
+
+const schema = Joi.object({
+  downloads_count: Joi.number().required(),
+  releases: Joi.array()
+    .items(
+      Joi.object({
+        version: Joi.string().required(),
+        released_at: Joi.string().required(),
+        info_json: Joi.object({
+          factorio_version: Joi.string().required(),
+        }).required(),
+      })
+    )
+    .min(1)
+    .required(),
+}).required()
+
+const queryParamSchema = Joi.object({
+  range: Joi.equal(''),
+}).required()
+
+// Factorio Mod portal API
+// @see https://wiki.factorio.com/Mod_portal_API
+class BaseFactorioModPortalService extends BaseJsonService {
+  async fetch({ modName }) {
+    const { releases, downloads_count } = await this._requestJson({
+      schema,
+      url: `https://mods.factorio.com/api/mods/${modName}`,
+      errorMessages: {
+        404: 'mod not found',
+      },
+    })
+
+    return {
+      downloads_count,
+      first_release: releases[0],
+      latest_release: releases[releases.length - 1],
+    }
+  }
+}
+
+// Badge for mod's latest updated version
+class FactorioModPortalLatestVersion extends BaseFactorioModPortalService {
+  static category = 'version'
+
+  static route = {
+    base: 'factorio-mod-portal/latest-version',
```

Comment:
Normally we use `/v` as the route for version badges - see https://github.com/badges/shields/blob/master/doc/badge-urls.md